signerv2/remote_signer_client.go

```diff
@@ -0,0 +1,107 @@
+package signerv2
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/Layr-Labs/eigensdk-go/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type RpcRequest struct {
+	JsonRPC string      `json:"jsonrpc"`
+	Method  string      `json:"method"`
+	Params  interface{} `json:"params"`
+	ID      int         `json:"id"`
+}
+
+type RemoteSignerClient interface {
+	SignTransaction(from common.Address, tx *types.Transaction) (*types.Transaction, error)
+}
+
+// RemoteSigner is a client for a remote signer
+// It implements Consensys Web3 Signer
+// Reference: https://docs.web3signer.consensys.io/reference/api/json-rpc
+type RemoteSigner struct {
+	url    string
+	client http.Client
+}
+
+func NewRemoteSignerClient(url string) RemoteSignerClient {
+	client := http.Client{}
+	return &RemoteSigner{client: client, url: url}
+}
+
+func (r RemoteSigner) SignTransaction(
+	from common.Address,
+	tx *types.Transaction,
+) (*types.Transaction, error) {
+	method := "eth_signTransaction"
+	id := 1
```

Comment:
maybe generate a unique id instead and include it in error msgs, such that if ever there's an error you can go to the web3signer logs and find the request easily

Reply:
7a6df19465ed37185f3719c841bd4ec3d2e9ddaf